ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -756,23 +757,56 @@ private void verifyReadFromFile(NavigableMap<MockId, NavigableSet<IndexValue>> r
     // read from file (unmapped) and verify that everything is ok
     Journal journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     IndexSegment fromDisk = createIndexSegmentFromFile(file, false, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, false, endOffset,
-        lastModifiedTimeInMs, resetKey);
-    verifyFind(referenceIndex, fromDisk);
-    verifyGetEntriesSince(referenceIndex, fromDisk);
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, false,
+        endOffset, lastModifiedTimeInMs, resetKey);
     // journal should contain all the entries
     verifyJournal(referenceIndex, journal);
     fromDisk.map(true);
 
     // read from file (mapped) and verify that everything is ok
     journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     fromDisk = createIndexSegmentFromFile(file, true, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, true, endOffset,
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, true,
+        endOffset, lastModifiedTimeInMs, resetKey);
+    // journal should not contain any entries
+    assertNull("Journal should not have any entries", journal.getFirstOffset());
+
+    // delete the bloom file
+    File BloomFile = new File(file.getParent(),
+        IndexSegment.generateIndexSegmentFilenamePrefix(startOffset) + BLOOM_FILE_NAME_SUFFIX);
+    BloomFile.delete();
+
+    // read from file (mapped) again and verify that everything is ok
+    journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
+    fromDisk = createIndexSegmentFromFile(file, true, journal);
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, true,
```

Comment:
before this add `assertTrue("Bloom file does not exist", bloomFile.exists())`

Reply:
I think you mean `assertFalse("Bloom file should not exist", bloomFile.exists())`, because the bloom file is already deleted. The statement is to ensure the file doesn't exist.

Reply:
Before this line, the new code you added creates a bloom filter, does it not?

Reply:
I see. I thought it was before line 779, ensuring the file didn't exist after deleting the bloom file. I will add the assert as you suggest.